src/libcrun/criu.c

```diff
@@ -218,6 +218,88 @@ libcrun_container_checkpoint_linux_criu (libcrun_container_status_t *status,
   return 0;
 }
 
+static int
+prepare_restore_mounts (runtime_spec_schema_config_schema *def,
+                        char *root,
+                        libcrun_error_t *err)
+{
+  int i;
+
+  /* Go through all mountpoints to be able to recreate missing mountpoints. */
+  for (i = 0; i < def->mounts_len; i++)
+    {
+      char *dest = def->mounts[i]->destination;
+      char *type = def->mounts[i]->type;
+      cleanup_close int root_fd = -1;
+      bool on_tmpfs = false;
+      int is_dir = 1;
+      size_t j;
+
+      /* cgroup restore should be handled by CRIU itself */
+      if (strcmp (type, "cgroup") == 0
+          || strcmp (type, "cgroup2") == 0)
+        continue;
+
+      for (j = 0; j < def->mounts_len; j++)
+        {
+          cleanup_free char *dest_loop = NULL;
+
+          xasprintf (&dest_loop, "%s/", def->mounts[j]->destination);
+          /* Check if the parent mount is on a tmpfs. CRIU restores
```

Comment:
I'd move this comment to before `for (j = 0, ...` as this is what the whole loop is for.

Reply:
changed